Data/These.hs

```diff
@@ -174,6 +179,24 @@ mapThese f _ (This  a  ) = This (f a)
 mapThese _ g (That    x) = That (g x)
 mapThese f g (These a x) = These (f a) (g x)
 
+-- | Case specific map.
+mapThese' :: (a -> c) -> (b -> d) -> (a -> b -> (c,d)) -> These a b -> These c d
```

Comment:
I'd need a motivating example

Reply:
I need to produce a value from `These` with different  logic for the `These` case. I have written this function a few times, and felt I needed to contribute it back.